### DISCUSSION TITLE
Practicum Submission | Generative AI for Financial Chatbots Workshop

### DISCUSSION THREAD

**Discussion by jeremiaaron** [onlyphantom/llm-python#40] [Practicum: Generative AI]
Submission by Jeremiah Aaron | jere
Google Colab Link: https://colab.research.google.com/drive/1GXcLIQEal0UQD2I_6Aw2hedSTib9cMqV?usp=sharing

**Comment by onlyphantom** [comment]
sorry for grading this so late, but well done! marking it a pass!

Edit: circling back to add, there are perhaps more specific, robust ways to write 
- test_output_parser_symbol_invalid
- test_output_parser_mcap_invalid

the idea is to check specifically if the symbl is not 4 letters long (BBRI, BBCA, ADRO etc) and that if market cap detected is lesser than 0. But in both tests you did `pytest.raises(OutputParserException)` which is not specifically targeting these edge cases!
